src/pages/events/EventSignIn.jsx

```diff
@@ -1,97 +1,132 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { useHistory } from "react-router-dom";
 
 import { useParams } from "react-router-dom";
 
 import "./Events.scss";
 
 export default function EventSignIn() {
+  const history = useHistory();
   const { event_id } = useParams();
-  const members = [
-    {
-      "id": 0,
-      "first_name": "Bob",
-      "last_name": "Johnson"
-    },
-    {
-      "id": 1,
-      "first_name": "Deep",
-      "last_name": "Kalra"
-    },
-    {
-      "id": 2,
-      "first_name": "Adam",
-      "last_name": "Mahood"
-    },
-    {
-      "id": 3,
-      "first_name": "Sasha",
-      "last_name": "Nayar"
-    },
-    {
-      "id": 4,
-      "first_name": "Arham",
-      "last_name": "Abidi"
+  const set = new Set();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [participants, setParticipants] = useState([]);
+  const [semesterId, setSemesterId] = useState("");

```

Comment:
participants and semesterId are only used in the useEffect and no where else, so there is no need to save them in state. Remove these